Mage/src/main/java/mage/abilities/effects/common/cost/SpellsCostReductionControllerEffect.java

```diff
@@ -32,15 +33,25 @@ public class SpellsCostReductionControllerEffect extends CostModificationEffectI
     private final DynamicValue amount;
     private final boolean upTo;
     private ManaCosts<ManaCost> manaCostsToReduce = null;
+    private final boolean convertToGeneric;
```

Comment:
Add docs like:
* `// if no more colored mana then allow to reduce generic cost, see CardUtil.adjustCost`